ChiTech/ChiMath/Quadratures/cylindrical_angular_quadrature.cc

```diff
@@ -0,0 +1,285 @@
+  }
+
+  //  --------------------------------------------------------------------------
+  //  curvilinear product quadrature : compute additional parametrising factors
+  //  --------------------------------------------------------------------------
+  InitializeParameters();
+
+  //  --------------------------------------------------------------------------
+  //  print
+  //  --------------------------------------------------------------------------
+  if (verbose)
+  {
+    chi_log.Log(LOG_0) << "map_directions" << std::endl;
+    for (const auto& dir : map_directions)
+    {
+      chi_log.Log(LOG_0) << "polar level " << dir.first << " : ";
+      for (const auto& q : dir.second)
+        chi_log.Log(LOG_0) << q << ", ";
+      chi_log.Log(LOG_0) << std::endl;
+    }
+    chi_log.Log(LOG_0) << "curvilinear product quadrature : cylindrical" << std::endl;
+    for (size_t k = 0; k < weights.size(); ++k)
+      chi_log.Log(LOG_0)
+        << "angle index " << k << ": weight = " << weights[k]
+        << ", (phi, theta) = (" << abscissae[k].phi << ", " << abscissae[k].theta << ")"
+        << ", omega = " << omegas[k]
+        << ", fac_diamond_difference = " << fac_diamond_difference[k]
+        << ", fac_streaming_operator = " << fac_streaming_operator[k] << std::endl;
+    const auto sum_weights =
+      std::accumulate(weights.begin(), weights.end(), 0.0);
+    chi_log.Log(LOG_0) << "sum(weights) = " << sum_weights << std::endl;
+  }
+}
+
+
+void
+chi_math::CylindricalAngularQuadrature::InitializeParameters()
+{
+  const auto pi_sum_q_weights = static_cast<double>(1);
+
+  fac_diamond_difference.resize(weights.size(), 1);
+  fac_streaming_operator.resize(weights.size(), 0);
+  for (size_t p = 0; p < map_directions.size(); ++p)
+  {
+    //  interface quantities initialised to starting direction values
+    double alpha_interface = 0;
+    double phi_interface = abscissae[map_directions[p].front()].phi;
+    std::vector<double> mu_interface(2, std::cos(phi_interface));
+
+    //  initialisation permits to forego start direction and final direction
+    for (size_t q = 1; q < map_directions[p].size()-1; ++q)
+    {
+      const auto k = map_directions[p][q];
+      const auto w_pq = weights[k];
+      const auto mu_pq = omegas[k].x;
+      const auto phi_pq = abscissae[k].phi;
+
+      alpha_interface -= w_pq * mu_pq;
+
+      phi_interface -= w_pq * pi_sum_q_weights;
+      mu_interface[0] = mu_interface[1];
+      mu_interface[1] = std::cos(phi_interface);
+
+      const auto mu = std::cos(phi_pq);
+      const auto tau = (mu - mu_interface[0]) / (mu_interface[1] - mu_interface[0]);
+
+      fac_diamond_difference[k] = tau;
+      fac_streaming_operator[k] = alpha_interface / (w_pq * tau) + mu_pq;
+    }
+  }
+}
+
+
+void
+chi_math::CylindricalAngularQuadrature::MakeHarmonicIndices(int scatt_order, int dimension)
+{
+  if (m_to_ell_em_map.empty())
+  {
+    if (dimension == 1)
+    {
+      for (unsigned int l = 0; l <= scatt_order; ++l)
```

Comment:
Again, we shouldn't use unsigned here.

Reply:
Variables that refer to the scattering order are now ```unsigned int scattering_order``` to be consistent throughout the code (c.f. pull request #134).